sql/analyzer/indexed_joins.go

```diff
@@ -651,13 +655,172 @@ func addHashJoins(m *memo.Memo) error {
 	})
 }
 
+type rangeFilter struct {
+	value, min, max                        *memo.ExprGroup
+	closedOnLowerBound, closedOnUpperBound bool
+}
+
+func getRangeFilters(filters []memo.ScalarExpr) (ranges []rangeFilter) {
```

Comment:
A useful way to document helper functions like this can be unit tests. They can help walk others through how to add additional cases or patches without having to step through a full query's debugger. `select_hint_tests.go` is maybe one example

Reply:
Done.